packages/api/src/utils/multipart/utils.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable */
+
+/**
+ * @param {string | number | boolean} s
+ */
+export function stringToArray(s) {
+  const utf8 = unescape(encodeURIComponent(s))
+  return Uint8Array.from(utf8, (_, i) => utf8.charCodeAt(i))
+}
+/**
+ * @param {number[]} a
+ */
+export function arrayToString(a) {
+  const utf8 = String.fromCharCode.apply(null, a)
+  return decodeURIComponent(escape(utf8))
+}
+/**
+ * @param {Uint8Array[]} arrays
+ */
+export function mergeArrays(...arrays) {
+  const out = new Uint8Array(
+    arrays.reduce((total, arr) => total + arr.length, 0)
+  )
+  let offset = 0
+  for (const arr of arrays) {
+    out.set(arr, offset)
+    offset += arr.length
+  }
+  return out
+}
+/**
+ * @param {string | any[]} a
+ * @param {string | any[] | Uint8Array} b
+ */
+export function arraysEqual(a, b) {
+  if (a.length !== b.length) {
+    return false
+  }
+  for (let i = 0; i < a.length; i++) {
+    if (a[i] !== b[i]) {
+      return false
+    }
+  }
+  return true
+}
```

Comment:
Please open an issue to pull this out as a dependency (or just do it 😉). It'll be useful for NFT.Storage also.

Reply:
Yes! I will open an issue and do this as a follow up to also PR NFT.storage

Reply:
https://github.com/web3-storage/web3.storage/issues/114